src/compiler/crystal/tools/formatter.cr

```diff
@@ -828,17 +828,23 @@ module Crystal
           accept element
         end
 
+        element_is_heredoc = !@lexer.heredocs.empty?
```

Comment:
I think this variable name is not correct. It should be named `has_heredoc_in_line` or `has_heredoc` as short.